KOPS.md

```diff
@@ -0,0 +1,130 @@
+When using hostNetwork, the port `8080` used by aws-encryption-provider conflicts with
+kube-apiserver which also requires the same port. To fix this, add `-health-port=:8083`
+to args section of pod spec above. Also change the port in `containerPort` and `livenessProbe`
+sections.
+
+#### Add /var/run/kmsplugin hostMount to api server spec
+Use kops lifecycle hook to run a script/container that can update the kube-apiserver 
+manifest (available at /etc/kubernetes/manifests) to add `/var/run/kmsplugin` as hostMount. Alternatively, you can mount a 
+directory that is already mounted by default e.g. `/srv/kubernetes/socket.sock`.
+
+#### Run aws-encryption-provider as static pod
+You need to have encryption provider running before kube-apiserver, and to do that you can
+use [static pods](https://kubernetes.io/docs/tasks/administer-cluster/static-pod/) functionality. For kops, static pod manifests are available at `/etc/kubernetes/manifests`. You can further use kops file assets functionality to drop 
+the pod spec file in that directory. 
+
+In the end, the kops specification for the file asset should look like the following:
+```yaml
+kind: Cluster
+spec:
+  fileAssets:
+  - name: aws-encryption-provider.yaml
+    ## Note if not path is specified the default path it /srv/kubernetes/assets/<name>
+    path: /etc/kubernetes/manifests/aws-encryption-provider.yaml
+    roles:
+    - Master
+    content: |
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: aws-encryption-provider
+        namespace: kube-system
+      spec:
+        containers:
+        - image: <image-of-aws-provider>
+          name: aws-encryption-provider
+          command:
+          - /aws-encryption-provider
+          - -key=<arn-of-kms-key>
+          - -region=<region-of-kms-key>
+          - -listen=/srv/kubernetes/socket.sock
+          - -health-port=:8083
+          ports:
+          - containerPort: 8083
+            protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 8083
+          volumeMounts:
+          - mountPath: /srv/kubernetes
+            name: kmsplugin
+        hostNetwork: true
+        volumes:
+        - name: kmsplugin
+          hostPath:
+            path: /srv/kubernetes
+            type: DirectoryOrCreate
+        nodeSelector:
+          dedicated: master
+        tolerations:
+        - key: dedicated
+          operator: Equal
+          value: master
+          effect: NoSchedule
+```
+Note: Here the plugin is mounted at `/srv/kubernetes/socket.sock`. Also, the above uses labels to make sure that the pod lives on all the same nodes as the kube-apiserver. The following is the kops specification to implement node labels for the master instance group to go with the above example:
+```yaml
+kind: InstanceGroup
+spec:
+  nodeLabels:
+    dedicated: master
+  role: Master
+```
+
+## Community, discussion, contribution, and support
```

Comment:
You can drop this and the remaining sections of this file

Reply:
Resolved in 0468159.